Lib/_android_support.py

```diff
@@ -0,0 +1,99 @@
+import io
+import sys
+from ctypes import CDLL, c_char_p, c_int
+
+
+# The maximum length of a log message in bytes, including the level marker and
+# tag, is defined as LOGGER_ENTRY_MAX_PAYLOAD in
+# platform/system/logging/liblog/include/log/log.h. As of API level 30, messages
+# longer than this will be be truncated by logcat. This limit has already been
+# reduced at least once in the history of Android (from 4076 to 4068 between API
+# level 23 and 26), so leave some headroom.
+MAX_BYTES_PER_WRITE = 4000
```

Comment:
Did you consider including the header in C, and passing the value to `init_streams`?

Reply:
This value is not in the public headers available when building Python – the reference in this comment is to an internal header of the logging library. 

Since the limit has varied between different Android versions in in the past, and there's no way to tell what it is at runtime, I wanted to leave a reasonable margin of safety, and making it a round 4000 simplifies the tests.

However, the ANDROID_LOG_INFO and ANDROID_LOG_WARN constants *are* in the public C header, so I've changed the code to pass those in.